src/main/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -204,4 +208,173 @@ public int draw() {
         }
         return normalizedHistogram;
     }
-}
+
+    /**
+     * Determines if a variant would be called under the particular conditions of a given total depth, alt depth,
+     * average base qualities, allele fraction of variant and log odds threshold necessary to call variant.
+     * @param totalDepth Depth at the site to be called, both alt and ref.
+     * @param altDepth Number of alt bases at this site.
+     * @param sumOfAltQualities Average Phred-scaled quality of bases
+     * @param alleleFraction Allele fraction we are attempting to detect
+     * @param logOddsThreshold Log odds threshold necessary for variant to be called
+     * @return
+     */
+     static boolean isCalled(final int totalDepth, final int altDepth, final double sumOfAltQualities, final double alleleFraction, final double logOddsThreshold) {
+        final double threshold = 10.0 * (altDepth * -Math.log10(alleleFraction) + (totalDepth - altDepth) * -Math.log10(1.0 - alleleFraction) + logOddsThreshold);
+
+        return sumOfAltQualities > threshold;
+    }
+
+    public TheoreticalSensitivity() {
+    }
+
+    /**
+     * Calculates the theoretical sensitivity with a given Phred-scaled quality score distribution at a constant
+     * depth.
+     * @param depth Depth to compute sensitivity at
+     * @param qualityDistribution Phred-scaled quality score distribution
+     * @param logOddsThreshold Log odd threshold necessary for variant to be called
+     * @param sampleSize sampleSize is the total number of simulations to run
+     * @param alleleFraction the allele fraction to evaluate sensitivity at
+     * @param randomSeed random number seed to use for random number generator
+     * @return Theoretical sensitivity for the given arguments at a constant depth.
+     */
+    public static double sensitivityAtConstantDepth(final int depth, final double[] qualityDistribution, final double logOddsThreshold, final int sampleSize, final double alleleFraction, final int randomSeed) {
+        final RouletteWheel qualityRW = new RouletteWheel(trimDistribution(qualityDistribution));
+        final Random randomNumberGenerator = new Random(randomSeed);
+        final BinomialDistribution bd = new BinomialDistribution(depth, alleleFraction);
+
+        // Calculate mean and deviation of quality score distribution to enable Gaussian sampling below.
+        final double averageQuality = IntStream.range(0, qualityDistribution.length-1).mapToDouble(n -> n * qualityDistribution[n]).sum();
```

Comment:
not an average

Reply:
use picard.util.MathUtil#mean and also picard.util.MathUtil#stddev(double[], int, int, double)